### PR TITLE
[MediaManager] Improve autoplay actions.

### PR DESCRIPTION
**Pull request by 78andyp**
## Description

Properly separate actions in MediaManager.
Add Win32 support.

## Motivation and context

Fix #27943.

The wiki is not clear about what happens when autoplay is not set:
image

Also fix failure of win32 to detect disc/removable storage insertions.
Note they are recognised if present when Kodi starts.

## How has this been tested?

Locally

## Screenshots (if appropriate):

Options:

None
image
Just shows the transient popup notification.

Play
image
For blurays (shows the simple menu - if selected)
For DVDs it just plays the disc

Browse
image
Shows the browse dialog

## Types of change

- [X] **Improvement** (non-breaking change which improves existing functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [X] All new and existing tests passed

### CODE REVIEW

File: system/settings/settings.xml [hunk 1]
```diff
@@ -420,11 +420,14 @@
...
     <category id="discs" label="14087" help="36193">
       <group id="1" label="446">
-        <setting id="dvds.autorun" type="boolean" label="14088" help="36194">
-          <requirement>HAS_OPTICAL_DRIVE</requirement>
-          <level>0</level>
-          <default>false</default>
-          <control type="toggle" />
+        <setting id="dvds.autoaction" type="integer" label="14088" help="36194">
+		  <requirement>HAS_OPTICAL_DRIVE</requirement>
+		  <level>0</level>
```
[the remark this answers is not in the document]
Comment (78andyp) [reply system/settings/settings.xml]: DOne

File: addons/resource.language.en_gb/resources/strings.po [hunk 2]
```diff
@@ -20126,7 +20129,7 @@ msgstr ""
 #. Description of setting with label #14088 "Play DVDs automatically"
```
[the remark this answers is not in the document]
Comment (78andyp) [reply addons/resource.language.en_gb/resources/strings.po]: Changed.

File: xbmc/Autorun.cpp [hunk 3]
```diff
@@ -110,7 +110,7 @@ bool CAutorun::PlayDisc(const std::string& path, const PlayDiscOptions& options)
   const AutoCDAction action =
       static_cast<AutoCDAction>(settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION));
   if (!options.bypassSettings && action != AutoCDAction::PLAY &&
-      !settings->GetBool(CSettings::SETTING_DVDS_AUTORUN))
+      !settings->GetBool(CSettings::SETTING_DVDS_AUTOACTION))
     return false;
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/Autorun.cpp]: Done

File: xbmc/Autorun.cpp [hunk 4]
```diff
@@ -202,7 +202,7 @@ bool CAutorun::RunDisc(IDirectory* pDir,
   {
     std::string hddvdname = "";
     CFileItemPtr phddvdItem;
-    bool bAutorunDVDs = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_DVDS_AUTORUN);
+    bool bAutorunDVDs = CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(CSettings::SETTING_DVDS_AUTOACTION);
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/Autorun.cpp]: Done

File: xbmc/Autorun.cpp [hunk 5]
```diff
@@ -431,7 +431,7 @@ bool CAutorun::RunDisc(IDirectory* pDir,
   // check video first
   if (!nAddedToPlaylist && !bPlaying &&
       (options.bypassSettings || CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
-                                     CSettings::SETTING_DVDS_AUTORUN)))
+                                     CSettings::SETTING_DVDS_AUTOACTION)))
   {
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/Autorun.cpp:439]: Done

File: xbmc/settings/Settings.cpp [hunk 6]
```diff
@@ -372,7 +372,10 @@ void CSettings::InitializeOptionFillers()
 {
   // register setting option fillers
 #ifdef HAS_OPTICAL_DRIVE
-  GetSettingsManager()->RegisterSettingOptionsFiller("audiocdactions", MEDIA_DETECT::CAutorun::SettingOptionAudioCdActionsFiller);
+  GetSettingsManager()->RegisterSettingOptionsFiller(
+      "audiocdactions", MEDIA_DETECT::CAutorun::SettingOptionAudioCdActionsFiller);
+  GetSettingsManager()->RegisterSettingOptionsFiller(
+      "videodiscactions", MEDIA_DETECT::CAutorun::SettingOptionVideoDiscActionsFiller);
 #endif
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/settings/Settings.cpp:379]: Done

File: xbmc/settings/Settings.h [hunk 7]
```diff
@@ -181,7 +181,7 @@ class CSettings : public CSettingsBase, public CSettingCreator, public CSettingC
   static constexpr auto SETTING_SUBTITLES_DOWNLOADFIRST = "subtitles.downloadfirst";
   static constexpr auto SETTING_SUBTITLES_TV = "subtitles.tv";
   static constexpr auto SETTING_SUBTITLES_MOVIE = "subtitles.movie";
-  static constexpr auto SETTING_DVDS_AUTORUN = "dvds.autorun";
+  static constexpr auto SETTING_DVDS_AUTOACTION = "dvds.autoaction";
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/settings/Settings.h]: Fixed.

File: xbmc/storage/MediaManager.cpp [hunk 8]
```diff
@@ -775,43 +777,87 @@ std::vector<std::string> CMediaManager::GetDiskUsage()
...
-void CMediaManager::OnStorageAdded(const MEDIA_DETECT::STORAGE::StorageDevice& device)
+void CMediaManager::OnStorageAdded(const STORAGE::StorageDevice& device)
 {
 #ifdef HAS_OPTICAL_DRIVE
-  const std::shared_ptr<CSettings> settings = CServiceBroker::GetSettingsComponent()->GetSettings();
-  if (settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION) !=
-          static_cast<int>(AutoCDAction::NONE) ||
-      settings->GetBool(CSettings::SETTING_DVDS_AUTORUN))
+  if (device.type == STORAGE::Type::OPTICAL)
   {
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/storage/MediaManager.cpp]: DOne

File: xbmc/Autorun.cpp [hunk 9]
```diff
@@ -605,3 +608,15 @@ void CAutorun::SettingOptionAudioCdActionsFiller(const SettingConstPtr& setting,
...
+void CAutorun::SettingOptionVideoDiscActionsFiller(const SettingConstPtr& setting,
+                                                   std::vector<IntegerSettingOption>& list,
+                                                   int& current)
+{
+  list.emplace_back(CServiceBroker::GetResourcesComponent().GetLocalizeStrings().Get(36462),
+                    static_cast<int>(AutoDVDAction::NONE));
+  list.emplace_back(CServiceBroker::GetResourcesComponent().GetLocalizeStrings().Get(36463),
+                    static_cast<int>(AutoDVDAction::BROWSE));
+  list.emplace_back(CServiceBroker::GetResourcesComponent().GetLocalizeStrings().Get(36464),
+                    static_cast<int>(AutoDVDAction::PLAY));
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/Autorun.cpp]: Fixed

File: system/settings/settings.xml [hunk 10]
```diff
@@ -420,11 +420,17 @@
...
-          <default>false</default>
-          <control type="toggle" />
+          <default>0</default> <!-- None -->
+          <constraints>
+            <options>videodiscactions</options>
+          </constraints>
+		  <updates>
+            <update type="change" />
+          </updates>
+		  <control type="list" format="string" />
```
[the remark this answers is not in the document]
Comment (78andyp) [reply system/settings/settings.xml]: Done

**Comment by 78andyp** [comment]
@kodiai review

File: xbmc/storage/MediaManager.cpp [hunk 11]
```diff
@@ -778,40 +780,84 @@ std::vector<std::string> CMediaManager::GetDiskUsage()
...
+        pInfo->IsAudio(1)}; // If it returns null, it is likely to be a protected video disc
+
+    if (isAudioDisc)
     {
-      CServiceBroker::GetJobManager()->AddJob(new CAutorunMediaJob(device.label, device.path), this,
-                                              CJob::PRIORITY_LOW);
+      const auto cdAutoAction{
+          static_cast<AutoCDAction>(settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION))};
+      CLog::LogF(LOGDEBUG, "Audio CD detected with path {}, auto action is {}", device.path,
+                 cdAutoAction);
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/storage/MediaManager.cpp]: Done

File: xbmc/storage/MediaManager.cpp [hunk 12]
```diff
@@ -778,40 +780,84 @@ std::vector<std::string> CMediaManager::GetDiskUsage()
...
+            TOAST_DISPLAY_TIME, false);
+      }
     }
     else
     {
-      if (device.type == MEDIA_DETECT::STORAGE::Type::OPTICAL)
+      const auto dvdAutoAction{
+          static_cast<AutoDVDAction>(settings->GetInt(CSettings::SETTING_DVDS_AUTOACTION))};
+      CLog::LogF(LOGDEBUG, "Video disc detected with path {}, auto action is {}", device.path,
+                 dvdAutoAction);
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/storage/MediaManager.cpp]: Done

File: xbmc/storage/MediaManager.cpp [hunk 13]
```diff
@@ -778,40 +780,84 @@ std::vector<std::string> CMediaManager::GetDiskUsage()
...
   {
-    if (settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION) ==
-        static_cast<int>(AutoCDAction::RIP))
+    const std::shared_ptr<CSettings> settings{
+        CServiceBroker::GetSettingsComponent()->GetSettings()};
+
+    CCdInfo* pInfo{GetCdInfo(device.path)};
+    bool isAudioDisc{
+        pInfo &&
+        pInfo->IsAudio(1)}; // If it returns null, it is likely to be a protected video disc
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/storage/MediaManager.cpp]: DOne

File: xbmc/settings/lib/SettingsManager.cpp [hunk 14]
```diff
@@ -1111,16 +1111,20 @@ bool CSettingsManager::LoadSetting(const TiXmlNode* node, const SettingPtr& sett
...
                                                         : StringUtils::Empty))
   {
-    m_logger->warn("unable to read value of setting \"{}\"", settingId);
-    return false;
+    // check if there are any updates to process
+    if (updates.empty())
+    {
+      m_logger->warn("unable to read value of setting \"{}\"", settingId);
+      return false;
+    }
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/settings/lib/SettingsManager.cpp]: DOne

**Comment by 78andyp** [comment]
@kodiai review

File: xbmc/settings/lib/SettingsManager.cpp [hunk 15]
```diff
@@ -1129,7 +1132,7 @@ bool CSettingsManager::LoadSetting(const TiXmlNode* node, const SettingPtr& sett
   if (!updated && isDefault)
     setting->Reset();
 
-  return true;
+  return !failedRead;
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/settings/lib/SettingsManager.cpp]: DOne

File: xbmc/application/ApplicationSettingsHandling.cpp [hunk 16]
```diff
@@ -184,6 +187,21 @@ bool CApplicationSettingsHandling::OnSettingUpdate(const std::shared_ptr<CSettin
...
+  {
+    auto intSetting = std::static_pointer_cast<CSettingInt>(setting);
+    if (oldSettingNode->FirstChild())
+    {
+      const std::string oldValue = oldSettingNode->FirstChild()->ValueStr();
+      if (oldValue == "true")
+        return intSetting->SetValue(static_cast<int>(AutoDVDAction::PLAY));
+      else if (oldValue == "false")
+        return intSetting->SetValue(static_cast<int>(AutoDVDAction::NONE));
+    }
```
[the remark this answers is not in the document]
Comment (78andyp) [reply xbmc/application/ApplicationSettingsHandling.cpp:207]: Done

**Comment by 78andyp** [comment]
@kodiai review

**Comment by 78andyp** [comment]
jenkins build this please

File: addons/resource.language.en_gb/resources/strings.po [hunk 17]
```diff
@@ -21615,7 +21618,25 @@ msgctxt "#36461"
...
+msgid "Play"
+msgstr ""
+
+#. Action on DVD/Bluray physical disc insertion - show a browse dialog
+#: system/settings/settings.xml
+msgctxt "#36464"
+msgid "Browse"
+msgstr ""
+
+#empty strings from id 36465 to 36499
```
Comment (garbear) [inline addons/resource.language.en_gb/resources/strings.po]: These duplicate existing strings?

Comment (78andyp) [reply addons/resource.language.en_gb/resources/strings.po]: Fair comment - tweaked, thanks.

**Comment by 78andyp** [comment]
jenkins build this please

**Comment by 78andyp** [comment]
Looking to merge soon unless objections.

File: addons/resource.language.en_gb/resources/strings.po [hunk 18]
```diff
@@ -8298,7 +8301,7 @@ msgstr ""
 
 #: system/settings/settings.xml
 msgctxt "#14088"
-msgid "Play DVDs automatically"
+msgid "Video disc insert action"
```
Comment (garbear) [inline addons/resource.language.en_gb/resources/strings.po]: It's better to nix a string and add one with a new ID than changing an existing string. We learned this once, but I forget when/why.

Comment (78andyp) [reply addons/resource.language.en_gb/resources/strings.po]: Thanks. Changed.

File: addons/resource.language.en_gb/resources/strings.po [hunk 19]
```diff
@@ -20229,10 +20234,10 @@ msgctxt "#36193"
 msgid "This category contains the settings for how DVD's, Blu-ray, & CD's are handled."
 msgstr ""
 
-#. Description of setting with label #14088 "Play DVDs automatically"
+#. Description of setting with label #14085 "Video disc insert action"
 #: system/settings/settings.xml
 msgctxt "#36194"
-msgid "Autorun DVD video when inserted in drive."
+msgid "Action to take when a video disc is inserted."
```
Comment (garbear) [inline addons/resource.language.en_gb/resources/strings.po]: This changes the semantics of the string, so I would use a new ID. Same for the following one.

File: addons/resource.language.en_gb/resources/strings.po [hunk 20]
```diff
@@ -8561,7 +8561,13 @@ msgctxt "#14128"
 msgid "Allow double refresh rates"
 msgstr ""
 
-#empty strings from id 14129 to 14186
+#. Action on DVD/Bluray physical disc insertion - show a browse dialog
+#: system/settings/settings.xml
+msgctxt "#14129"
+msgid "Browse"
```
Comment (garbear) [inline addons/resource.language.en_gb/resources/strings.po]: This string already exists?

Comment (78andyp) [reply addons/resource.language.en_gb/resources/strings.po]: It does as #1024 - but as the meaning isn't clear I thought it best to add a new string (in case there are multiple words for browse in a language).

Comment (garbear) [reply addons/resource.language.en_gb/resources/strings.po]: 1024 has no context, so it's not like the translators have any opportunity for words of different contexts. I would research 1024 and add a context comment to it depending on how it's used. There's a cost to new strings (effort, wide lack of translations) so using an existing string would be beneficial if possible.

Comment (78andyp) [reply addons/resource.language.en_gb/resources/strings.po]: OK. It looks like one of the files doesn't exist any more and I've added comments to explain the use cases.

**Review by garbear — APPROVED** [review]
No problems that I can see.

File: system/settings/settings.xml [hunk 21]
```diff
@@ -425,11 +425,17 @@
     </category>
     <category id="discs" label="14087" help="36193">
       <group id="1" label="446">
-        <setting id="dvds.autorun" type="boolean" label="14088" help="36194">
+        <setting id="dvds.autorun" type="integer" label="14085" help="36416">
```
Comment (garbear) [inline system/settings/settings.xml]: Will is be a problem loading the old boolean parameter when expecting an int?

Comment (78andyp) [reply system/settings/settings.xml]: That's what the second commit is for - there was some missing functionality it the settings code to allow upgrades of v2 settings.

**Comment by CrystalP** [comment]
As mentioned in my other PR #28349 I started looking at the settings update problem as well for a different use case.
I think it deserves its own PR and discussion;  it's a part of the code that sees little use, some parts don't make sense to me.
I have something almost ready to propose, in multiple commits so that ideas can easily be removed/added and satisfy the needs of both PRs.

**Comment by 78andyp** [comment]
> As mentioned in my other PR #28349 I started looking at the settings update problem as well for a different use case. I think it deserves its own PR and discussion; it's a part of the code that sees little use, some parts don't make sense to me. I have something almost ready to propose, in multiple commits so that ideas can easily be removed/added and satisfy the needs of both PRs.

Cool, thanks. I can always remove the second commit from this.

**Comment by CrystalP** [comment]
The motivation is "The wiki is not clear about what happens when autoplay is not set:" yet there is no explanation in the PR either ;-) and it introduces additional options.

**Comment by 78andyp** [comment]
> The motivation is "The wiki is not clear about what happens when autoplay is not set:" yet there is no explanation in the PR either ;-) and it introduces additional options.

Fair point! I've added images to illustrate the 3 options.

**Comment by 78andyp** [comment]
@kodiai review

**Comment by 78andyp** [comment]
@kodiai - you have said not approved but provided no issues. Why not approve?

**Comment by 78andyp** [comment]
@kodiai - you are not talking about this PR?? Please review this PR carefully.

**Comment by 78andyp** [comment]
@kodiai you are hallucinating - items 2-5 are different PRs.

**Comment by 78andyp** [comment]
@thexai - could I trouble you to look at the second commit and the correctness of the win32 changes. Thanks.

File: xbmc/windowing/windows/WinEventsWin32.cpp [hunk 22]
```diff
@@ -790,31 +790,33 @@ LRESULT CALLBACK CWinEventsWin32::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, L
...
-                  share.strPath = strdrive;
-                  share.strName = share.strPath;
-                  CServiceBroker::GetMediaManager().RemoveAutoSource(share);
-                }
+                CLog::LogF(LOGDEBUG, "Drive {} Media has arrived.", strdrive);
+                CWin32StorageProvider::QueueStorageEvent(
+                    CWin32StorageProvider::StorageEventType::ADDED, device);
               }
               else
-                CWin32StorageProvider::SetEvent();
```
Comment (CrystalP) [inline xbmc/windowing/windows/WinEventsWin32.cpp:817]: seems this path is not handled anymore?

The PR touches fragile code that caused regressions every time we tried to improve it many years ago. Do you have Windows 10 and 8 computers to verify your changes? At least 7 and XP are not factors anymore.

Comment (78andyp) [reply xbmc/windowing/windows/WinEventsWin32.cpp:817]: Does a VM count??

Comment (78andyp) [reply xbmc/windowing/windows/WinEventsWin32.cpp:817]: Just tested in a Win 8.1 and Win 10 VM. Works fine in both (disc insertions recognised once Kodi is running, autorun settings respected, can eject). USB sticks fine as well.

Comment (78andyp) [reply xbmc/windowing/windows/WinEventsWin32.cpp:817]: > seems this path is not handled anymore?
> 
No - every DBT_DEVTYP_VOLUME event — optical or USB — gets a fully-populated StorageDevice and flows through PumpDriveChangeEvents() - as opposed to just flagging something has changed - which is what this did.

Comment (CrystalP) [reply xbmc/windowing/windows/WinEventsWin32.cpp:817]: I was probably thinking about hotplugging a SATA drive.

Comment (78andyp) [reply xbmc/windowing/windows/WinEventsWin32.cpp:817]: I think it still triggers DBT_DEVICEARRIVAL/DBT_DEVTYP_VOLUME so will go down the USB route, just not show up in GetRemovableDrives().
I'll add it back in anyway, can't do any harm.

**Comment by 78andyp** [comment]
If #28402 isn't going in then I'll bring back my targetted settings patch for this.

**Comment by 78andyp** [comment]
Rebased.

**Comment by 78andyp** [comment]
Rebased

**Comment by CrystalP** [comment]
needs to modify the existing setting with the new framework - instructions in #28402. The first update has some legwork though, couldn't push files with only placeholders. I plan to rework 21881 if you want me to do it.

**Comment by 78andyp** [comment]
Hi @CrystalP 
I've had a go - but I must be doing something wrong as the setting gets changed in `guisettings.xml` but the value is not preserved (true -> 0, instead of 1).
I put a breakpoint in the CSettingsMigrationToV3 and it never triggers.

In the log:
```
2026-08-08 18:17:13.027 T:35448 warning <CSettingsManager>: missing version attribute
2026-08-08 18:17:13.027 T:35448   debug <general>: CSettingsMigration::UpdateXMLSettings: upgrading settings from version 0 to 3
2026-08-08 18:17:13.028 T:35448   debug <general>: CSettingsMigration::UpdateXMLSettings: no migration steps available from version 0 to 3
```

**Comment by CrystalP** [comment]
Ah yes I wrote the test cases with multiple versions and there is a corner case for a single version... trivial fix. I'll do that.

The missing version attribute likely comes from the advancedsettings.xml, you should have another similar log trace, but recognizing the jump from 2 to 3.

**Comment by CrystalP** [comment]
Actually no the case I was thinking about is taken care of. Will build your PR and take it from there.

**Comment by 78andyp** [comment]
Thanks

File: system/settings/settings.xml [hunk 23]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8" ?>
-<settings version="1">
+<settings version="3">
```
Comment (CrystalP) [inline system/settings/settings.xml]: I don't think this is used anywhere.  It could actually be nice to bump version like this rather than through a constant in code. Will think about it, in the meantime please revert.

**Comment by CrystalP** [comment]
It's an embarrassing bug in the template code I provided...

**Comment by 78andyp** [comment]
This fixes the constructor not being called:
```
CSettingsMigration::CSettingsMigration()
  : CSettingsMigration(StepList{
        std::make_shared<CSettingsMigrationToV3>(),
    })
{
}
```

But I'm still struggling with the setting names - this doesn't work - and I feel it should do:
```
    constexpr std::string_view oldSettingId = "dvds.autorun";
    constexpr std::string_view newSettingId = "dvds.autoaction";
```

**Comment by CrystalP** [comment]
Yes. I think this is a bit tidier:
```
namespace
{
CSettingsMigration::StepList BuildMigrationSteps()
{
  return {
      std::make_shared<CSettingsMigrationToV3>(),
  };
}
} // namespace

CSettingsMigration::CSettingsMigration() : CSettingsMigration(BuildMigrationSteps())
{
}
```

> But I'm still struggling with the setting names - this doesn't work - and I feel it should do:
> 
> ```
>     constexpr std::string_view oldSettingId = "dvds.autorun";
>     constexpr std::string_view newSettingId = "dvds.autoaction";
> ```

Correct, that's what they have to be for this PR.

**Comment by 78andyp** [comment]
Odd. I've made your amendments and tried again and it works... true -> 1.
Thanks for the help :)

**Comment by 78andyp** [comment]
jenkins build this please

**Comment by CrystalP** [comment]
why odd? those are the setting names.
There are other settings with a name that starts with videolibrary.

The migration function should not be in a header imo.

**Comment by 78andyp** [comment]
Odd as I tried with those names and my constructor fix and whilst the constructor was called it failed saying couldn't find the old name. But tried again with yours and it works....

Agree about header - looking at the instructions again I confused myself:
image

Originally the only SettingsMigrationStep (note the Step) file I could find was the ISettingsMigrationStep.h (and it said .cpp/.h) - so I put it in there and it didn't complie so I then moved it to SettingsMigration.h

I'll change it.

**Comment by CrystalP** [comment]
That's because SettingsMigrationStep.h/.cpp have to be created :-) I couldn't really do it with no content inside.

I can do a follow up on what you have here, in light of 21881 I can see a refactor already in the cards. And the log messages about inexistent setting in advancedsettings.xml will probably have to be shut up, as having settings override there is the exception rather than the norm.

edit: I mean you move the function to the .cpp, I can take care of the new file creation.

**Comment by 78andyp** [comment]
Perfect - already moved to the cpp with the last push. Just waiting for it to build.
Sounds like a plan.

**Comment by 78andyp** [comment]
jenkins build this please

File: system/settings/settings.xml [hunk 24]
```diff
@@ -433,11 +433,17 @@
...
+        <setting id="dvds.autoaction" type="integer" label="14085" help="36440">
           <requirement>HAS_OPTICAL_DRIVE</requirement>
           <level>0</level>
-          <default>false</default>
-          <control type="toggle" />
+          <default>0</default> <!-- None -->
+          <constraints>
+            <options>videodiscactions</options>
+          </constraints>
+          <updates>
```
Comment (CrystalP) [inline system/settings/settings.xml:443]: that piece was not necessary anymore

Comment (78andyp) [reply system/settings/settings.xml:443]: Worth a PR to tidy up??
